src/poprox_storage/repositories/articles.py

```diff
@@ -228,6 +241,15 @@ def store_articles(self, articles: list[Article], *, mentions=False, progress=Fa
 
         return failed
 
+    def store_candidates(self, articles: list[Article], date: date):
+        candidates_table = self.tables["candidate_articles"]
+        insert_stmt = (
+            insert(candidates_table)
+            .values([{"article_id": article.article_id, "date": date} for article in articles])
```

Comment:
I don't love `date` here to identify which newsletter batch we're talking about. But I like it better than every way I can see to fix it.

Realistically it's fine so long as we make sure never to send two newsletters in one calendar day.

Reply:
One thing we could do is give each candidate set its own id and then link newsletters to the specific candidate set they used